src/main/java/com/uid2/admin/monitoring/DataStoreMetrics.java

```diff
@@ -21,4 +26,23 @@ public static void addDataStoreMetrics(String dataType, IMetadataVersionedStore
                 .description("version from metadata of a data store")
                 .register(globalRegistry);
     }
+    public static void addDataStoreSnowflakeMetrics(String dataType, RotatingServiceLinkStore serviceLinkStore, RotatingServiceStore serviceStore) {
+        Gauge
+                .builder("uid2_data_store_entry_count", () -> {
+                    try {
+                        // Warning: this downloads metadata from the underlying remote data store
```

Comment:
I thought it was the call to `getMetadata` in the method above that downloads metadata. Where are we downloading metadata in this method?

Reply:
That's true, I haven't loaded the content yet. I'll fix that!